CRM/Queue/BAO/Queue.php

```diff
@@ -20,6 +20,39 @@
  */
 class CRM_Queue_BAO_Queue extends CRM_Queue_DAO_Queue {
 
+  /**
+   * Get a list of valid statuses.
+   *
+   * This error-mode kicks in if there are no hooks and no retries left.
+   *
+   * @return string[]
+   */
+  public static function getStatuses($context = NULL) {
+    return [
+      'active' => ts('Active'),
+      // 'complete' => ts('Complete'),
+      // 'draft' => ts('Draft'),
+      // 'aborted' => ts('Aborted'),
+    ];
+  }
+
+  /**
+   * Get a list of valid error modes.
+   *
+   * This error-mode kicks in if there are no hooks and no retries left.
+   *
+   * @return string[]
+   */
+  public static function getErrorModes($context = NULL) {
+    return [
+      'log' => ts('Log details of failed task'),
```

Comment:
So - is this a multple option - ie `log` is in combo with `drop` or `abort` isn't it?

Reply:
To update the Github record - It wasn't really intended as multiple-option, and this draft had conflated two dimensions. We discussed this a bit more on MM and found two distinct dimensions of the error policy. 

With the latest revision, it addresses the two dimensions as:

1. __Where to complain about unhandled errors__: This sends complaints to `Civi::log('queue')`. (But it may be good to develop other destinations.)
2. __Whether unhandled errors are fatal__: This allows fatal error-handling (`error=>abort`) and non-fatal (`error=>delete`).